Add XML output support

guyzyl opened the pull request:
Relates to #81.
Doesn't completely solve it since Azure DevOps expects the XML to be in JUnit format, but it takes the project one step closer towards that goal.

guyzyl committed:
Update evaluator.go

Converted FileNameRuleMapper to type

guyzyl committed:
Added XML output support

guyzyl committed:
Update main_test.go

Added XML output CLI test

myishay reviewed (commented):
Thank you for your contribution!

I played a bit with the xml output and it seems like the items in `FileNameRuleMapper` are not applying xml format rules. 

I created a basic policy check to our `k8s-demo.yaml` file using only `Ensure workload has valid label values` rule and this was the xml output result:
```
<?xml version="1.0" encoding="UTF-8"?>
<FormattedOutput>
	<EvaluationResults>
		<FileNameRuleMapper>
			<./internal/fixtures/kube/k8s-demo.yaml>
				<9>
					<Rule>
						<ID>9</ID>
						<Name>Ensure workload has valid label values</Name>
						<FailSuggestion>Incorrect value for key(s) under `labels` - the vales syntax is not valid so the Kubernetes engine will not accept it</FailSuggestion>
						<OccurrencesDetails>
							<MetadataName>rss-site</MetadataName>
							<Kind>Deployment</Kind>
						</OccurrencesDetails>
					</Rule>
					<Rule>
						<ID>9</ID>
						<Name>Ensure workload has valid label values</Name>
						<FailSuggestion>Incorrect value for key(s) under `labels` - the vales syntax is not valid so the Kubernetes engine will not accept it</FailSuggestion>
						<OccurrencesDetails>
							<MetadataName>rss-site</MetadataName>
							<Kind>Deployment</Kind>
						</OccurrencesDetails>
					</Rule>
				</9>
			</./internal/fixtures/kube/k8s-demo.yaml>
		</FileNameRuleMapper>
		<Summary>
			<TotalFailedRules>1</TotalFailedRules>
			<FilesCount>1</FilesCount>
			<TotalPassedCount>0</TotalPassedCount>
		</Summary>
	</EvaluationResults>
	<EvaluationSummary>
		<ConfigsCount>1</ConfigsCount>
		<RulesCount>15</RulesCount>
		<FilesCount>1</FilesCount>
		<PassedYamlValidationCount>1</PassedYamlValidationCount>
		<PassedK8sValidationCount>1</PassedK8sValidationCount>
		<PassedPolicyCheckCount>0</PassedPolicyCheckCount>
	</EvaluationSummary>
</FormattedOutput>
```

A few things that I find a bit problematic:
- xml tags should not contain `/` character as the tag type, as in the filename tag for ecample: `<./internal/fixtures/kube/k8s-demo.yaml>`. A good solution would be changing the tag to something like `<File filename="./internal/fixtures/kube/k8s-demo.yaml">`
- xml tags should not start with a number, as in rule id tag for example: `<9>`. A good solutions would be remving this tag since it encapsulates a `<Rule>` tag anyway.


Also, for some reason I didn't check yet, I see 2 appearances of rule id 9 in the tested file, whereas only 1 appearance should be the correct output (you can look at the yaml/json/text output to see how it should look like). Seems like a bug to me 🐛.

@guyzyl  WDYT?

guyzyl committed:
Update printer.go

Removed integer rule id tags

guyzyl committed:
Update printer.go

Fixed issue where rule is outputted twice

guyzyl committed:
Update printer.go

Changed file tag to include path as attribute instead of tag name + improved var naming

guyzyl commented:
@myishay I totally agree with all of your comments and I pushed fixed for all of them:
- Removed the rule integer id tag (ie: <9>)
- Changed `filepath` to be an attribute inside a `File` tag
- Fixed issue where the rule is outputted twice
- Improved var naming on the way :)

myishay reviewed (approved):
I'm good with the changes, and I also liked the naming improvements!
